kunta-api-server/src/main/java/fi/otavanopisto/kuntaapi/server/persistence/dao/OrganizationSettingDAO.java

```diff
@@ -0,0 +1,82 @@
+package fi.otavanopisto.kuntaapi.server.persistence.dao;
+
+import javax.enterprise.context.Dependent;
+import javax.persistence.EntityManager;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+
+import fi.otavanopisto.kuntaapi.server.persistence.model.OrganizationSetting;
+import fi.otavanopisto.kuntaapi.server.persistence.model.OrganizationSetting_;
+
+/**
+ * DAO class for OrganizationSetting entity
+ * 
+ * @author Otavan Opisto
+ */
+@Dependent
+public class OrganizationSettingDAO extends AbstractDAO<OrganizationSetting> {
+
+  private static final long serialVersionUID = 8710294058847591299L;
+
+  /**
+   * Creates new OrganizationSetting entity
+   * 
+   * @param key key of setting. Must by unique among the organization
+   * @param value setting value. Not nullable.
+   * @param organizationKuntaApiId Kunta API id of organization
+   * 
+   * @return created OrganizationSetting entity
+   */
+  public OrganizationSetting create(String key, String value, String organizationKuntaApiId) {
+    OrganizationSetting organizationSetting = new OrganizationSetting();
+    
+    organizationSetting.setKey(key);
+    organizationSetting.setValue(value);
+    organizationSetting.setOrganizationKuntaApiId(organizationKuntaApiId);
+    
+    return persist(organizationSetting);
+  }
+
+  /**
+   * Finds organization setting by key and organizationIdentifier
+   * 
+   * @param key setting key
+   * @param organizationKuntaApiId Kunta API id of organization
+   * @return found setting or null if non found
+   */
+  public OrganizationSetting findByKeyAndOrganizationKuntaApiId(String key, String organizationKuntaApiId) {
+    EntityManager entityManager = getEntityManager();
+
+    CriteriaBuilder criteriaBuilder = entityManager.getCriteriaBuilder();
+    CriteriaQuery<OrganizationSetting> criteria = criteriaBuilder.createQuery(OrganizationSetting.class);
+    Root<OrganizationSetting> root = criteria.from(OrganizationSetting.class);
+    criteria.select(root);
+    criteria.where(
+      criteriaBuilder.and(
+        criteriaBuilder.equal(root.get(OrganizationSetting_.key), key),
+        criteriaBuilder.equal(root.get(OrganizationSetting_.organizationKuntaApiId), organizationKuntaApiId)
+      )
+    );
+    
+    return getSingleResult(entityManager.createQuery(criteria));
+  }
+  
+  /**
+   * Updates organization setting value
+   * 
+   * @param setting organization setting
+   * @param value new value
+   * @return updated organization setting
+   */
+  public OrganizationSetting updateValue(OrganizationSetting setting, String value) {
+    setting.setValue(value);
+    return persist(setting);
+  }
+  
+  @Override
+  public void delete(OrganizationSetting setting) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this method to simply inherit it. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1185)